File: 1250-longest-common-subsequence/longest-common-subsequence.cpp
class Solution {
public:
    int longestCommonSubsequence(string a, string b) {
        int n= (int)a.length(), m= (int)b.length();
        int dp[n+1][m+1];
        memset(dp, 0, sizeof(dp));
        for(int i=1;i<=n;++i){
            for(int j=1;j<=m;++j){
                if(a[i-1]==b[j-1]){
                    dp[i][j]=1+dp[i-1][j-1];
                }
                else dp[i][j]=max(dp[i-1][j], dp[i][j-1]);
            }
        }
        return dp[n][m];
    }
};

File: 1250-longest-common-subsequence/longest-common-subsequence.java
class Solution {
    public int longestCommonSubsequence(String text1, String text2) {
        int n1= text1.length(), n2= text2.length(), res;
        int[][] lcs= new int[n1+1][n2+1];
        for(int i=1;i<=n1;i++){
            for(int j=1;j<=n2;j++){
                if(text1.charAt(i-1)==text2.charAt(j-1)){
                    lcs[i%2][j]=lcs[1-i%2][j-1]+1;
                }
                else{
                    lcs[i%2][j]= Math.max(lcs[1-i%2][j],lcs[i%2][j-1]);
                }
            }
        }
        return lcs[n1%2][n2];
    }
}